core/startup/config_file_watcher.go

```diff
@@ -30,7 +30,7 @@ func ReadApiKeysJson(configDir string, appConfig *config.ApplicationConfig) erro
 }
 
 func ReadExternalBackendsJson(configDir string, appConfig *config.ApplicationConfig) error {
-	fileContent, err := os.ReadFile(path.Join(configDir, "external_backends.json"))
+	fileContent, err := os.ReadFile(filepath.Join(configDir, "external_backends.json"))
 	if err != nil {

```

Comment:
one of the errors that is likely to happen is the file not existing, we should catch that with something like:
```
	if err != nil {
		if os.IsNotExist(err) {
			return nil
		}
		return err
	}
```